Projects/Domain/Sources/Services/SubredditService.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Created by Igor Tarasenko on 2019-02-07.
+// Licensed under the MIT license
+//
+
+import Data
+import Foundation
+import RxSwift
+
+public protocol SubredditService {
+    func getPosts(for subredditName: String) -> Observable<[Post]>
+}
+
+public protocol SubredditServiceContainer {
+    var subredditService: SubredditService { get }
+}
+
+public final class DefaultSubredditService: SubredditService {
+    private let dataSource: SubredditDataSource
+
+    public init(dataSource: SubredditDataSource) {
+        self.dataSource = dataSource
+    }
+
+    public func getPosts(for subredditName: String) -> Observable<[Post]> {
+        return dataSource
+            .fetchPosts(for: subredditName)
+            .asObservable()
+            .map { $0.data.children?.map(DefaultSubredditService.convertPostContainerToDomainPost) }
+            .map { $0 ?? [] }
```

Comment:
Not sure I like that way of dealing with optionality. Why not filter them out instead (e.g. with `flatMap`; preferably using something like RxOptional or writing our own extension)?

Reply:
I still want to receive `onNext` event, even if there are no results. Don't focus on service right now; this is just a dumb implementation to have something to work with.